src/components/scenes/Staking/EarnScene.tsx

```diff
@@ -0,0 +1,168 @@
+  const { navigation } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatch(account, 'currencyWallets')
+  const wallets = Object.values(currencyWallets)
+
+  const [stakePolicyMap, setStakePolicyMap] = React.useState<StakePolicyMap>()
+
+  const [displayWallets, setDisplayWallets] = React.useState<EdgeCurrencyWallet[]>(wallets)
+  const [positionWallets, setPositionWallets] = React.useState<EdgeCurrencyWallet[]>([])
+
+  const [isPortfolioSelected, setIsPortfolioSelected] = React.useState(false)
+  const [isLoading, setIsLoading] = React.useState(true)
+
+  const handleSelectEarn = useHandler(() => setIsPortfolioSelected(false))
+  const handleSelectPortfolio = useHandler(() => setIsPortfolioSelected(true))
+
+  useAsyncEffect(
+    async () => {
+      if (stakePolicyMap != null) return
+
+      const positionWallets = []
+      const policyMap: StakePolicyMap = {}
+
+      for (const wallet of wallets) {
+        // Get all available stake policies
+        const { pluginId } = wallet.currencyInfo
+        if (SPECIAL_CURRENCY_INFO[pluginId]?.isStakingSupported === true && ENV.ENABLE_STAKING) {
+          // For each wallet
+          const stakePolicyPositions: StakePolicyPosition[] = []
+
+          try {
+            const stakePlugins = await getStakePlugins(pluginId)
+            for (const stakePlugin of stakePlugins) {
+              const stakePolicies = stakePlugin.getPolicies({ wallet, currencyCode: wallet.currencyInfo.currencyCode })
+
+              // Check if there's open positions
+              for (const stakePolicy of stakePolicies) {
+                const stakePosition = await stakePlugin.fetchStakePosition({ stakePolicyId: stakePolicy.stakePolicyId, wallet, account })
+                stakePolicyPositions.push({ stakePolicy, stakePosition })
+
+                const allocations = getPositionAllocations(stakePosition)
+                const { staked, earned, unstaked } = allocations
+                if ([...staked, ...earned, ...unstaked].some(positionAllocation => !zeroString(positionAllocation.nativeAmount))) {
+                  positionWallets.push(wallet)
+                }
+              }
+            }
+
+            policyMap[wallet.id] = { stakePolicyPositions, stakePlugins }
+            setStakePolicyMap({ ...policyMap })
+            setPositionWallets(positionWallets)
+          } catch (e) {
+            console.error(String(e))
+          }
+        }
+      }
+      setIsLoading(false)
+    },
+    [],
+    'EarnScene'
+  )
+
+  // Filter wallets based on isPortfolioSelected
+  React.useEffect(() => {
+    if (stakePolicyMap == null) return
+
+    if (isPortfolioSelected) {
+      setDisplayWallets(positionWallets)
+    } else {
+      setDisplayWallets(wallets)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isPortfolioSelected, stakePolicyMap])
```

Comment:
The `displayWallets` can be a plain variable, not a `useState`. Just do`const displayWallets = !stakePolicyMap ? [] : isPortfolioSelected ? positionWallets : wallets`, and it should have the same result.